### PR TITLE
Theme Tiers: Add Sensei and Community Tiers

### PR DESCRIPTION
**Pull request by Copons**
Related to D138731-code

## Proposed Changes

This PR introduces the Sensei and Community theme tiers already added to the Showcase site and added to the API in D138731-code.

To ensure the new tiers are not used in the Showcase filtering system, we are inverting how the filters are built: instead of using the API as source of truth, we are using the Calypso counterpart, allowing for a granular control over which tiers are displayed as filters, and which are there for other reasons (badges, feature checks, event tracking, or even simply consistency).

### The Sensei Tier

This change aims to track the engagement and revenue of all themes. Currently, the Sensei tier is not provisioned by the API, and Sensei themes (only one for now: Course) are not tracked correctly, recording a `null` tier.

### The Community Tier

The Community tier, for example, is added for consistency and future-proofing. Since Dotorg themes are retrieved via the Dotorg API, Calypso might not have the tier definitions in state (which is provided by the Dotcom themes API). I've hardcoded a barebone tier information to the normalized Dotorg theme objects, so that it can at least be used for tracking.

In the future, we should update how tiers are added to the Calypso state. Instead of piggybacking from other existing `/themes` and `/theme` endpoints, we should have a standalone endpoint for this purpose only.
For the sake of simplicity, let's start with a hardcode.

## Testing Instructions

* Apply D138731-code and sandbox the API.
* Select a free test site for convenience.
* Enter the Theme Showcase (`/themes`) and search for a Sensei theme (e.g Course).
* Open the theme and click on the upsell banner.
* Ensure the `calypso_theme_upsell_plan_click` event is recorded with `theme_tier: "sensei"`.
* Now search for a Community theme (e.g. Astra), open the theme and click on the upsell banner.
* Ensure the `calypso_theme_upsell_plan_click` event is recorded with `theme_tier: "community"`.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by xavier-lc** [comment]
> * Ensure the `calypso_theme_upsell_plan_click` event is recorded with `theme_tier: "sensei"`.

How can I check this? Looking for a request to a certain endpoint?

**Comment by Copons** [comment]
> > * Ensure the `calypso_theme_upsell_plan_click` event is recorded with `theme_tier: "sensei"`.
> 
> How can I check this? Looking for a request to a certain endpoint?

Ooh, good reminder of not assuming everyone has been working on Calypso for all their adult life! 😅 

Here you go: https://github.com/Automattic/wp-calypso/blob/04dbeab04afc6cf7f461f026f083d59d1034ee1c/client/lib/analytics/README.md

tl;dr
1. Run `localStorage.setItem('debug', 'calypso:analytics');` in the console.
2. Reload.
3. For convenience, filter the console for the event you want to observe (`calypso_theme_upsell_plan_click`).
4. Maybe check "Preserve log" so that the log persists after full page changes (IIRC opening the Checkout is a full page change, so it's useful in this case).
5. Follow the steps and observe the "Verbose" log. You should see something like this:

Screenshot 2024-02-20 at 15 43 45


Note that that banner is different and therefore triggers a different event for different type of themes.
For example, it's `calypso_banner_cta_click` for Personal and Premium.

**Comment by mmtr** [comment]
> How can I check this? Looking for a request to a certain endpoint?

In my experience the local storage debug mode is a bit buggy, so I ended up just monitoring the `t.gif` requests in the network tab of my browser devtools:

Screenshot 2024-02-21 at 11 40 37

These are the requests we trigger to actually track an event.

Similarly, you can also use the Vigilante browser extension to observe the Tracks events: p7H4VZ-3cB-p2

image

**Review by mmtr — APPROVED** [review]

**Comment by xavier-lc** [comment]
@Copons @mmtr thanks for sharing!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11384876).
